### ISSUE TITLE
How to use PositionEmbeddingSine standalone

### ISSUE THREAD

**Issue by liqi6811** [facebookresearch/detr#234]
Hi team, 

I am trying to learn everything step by step, and get an idea of details of each step. 

For PositionEmbeddingSine, I used below code to test what is the output: 

=============
PES = PositionEmbeddingSine()
print(PES)
PES = PES.to(device) 
for i, (inputs, labels) in enumerate(train_load):
    inputs, labels = Variable(inputs), Variable(labels.squeeze(1))       
    inputs, labels = inputs.to(device), labels.long().to(device)

    outputs = PES(inputs.float())  
=============

I got error as below: 
![image](https://user-images.githubusercontent.com/48280760/93452053-4c6f0980-f91b-11ea-8dae-56c3867d65b5.png)

so, I tried to see how does NestedTensor works, I used below code but stuck at what to pass as "mask: Optional[Tensor]"

![image](https://user-images.githubusercontent.com/48280760/93452202-7de7d500-f91b-11ea-907b-a46be874b055.png)

Thank you very much. 

Ling

**Comment by zhiqwang** [comment]
If you only want to use `PositionEmbeddingSine`, there is no need to set the object `NestedTensor` as the inputs. Something like

```python
class PositionEmbeddingSine(nn.Module):
    def __init__(...):
        ...

    def forward(self, tensor_input: torch.Tensor, mask: Optional[torch.Tensor] = None):
        ...
```

BTW, the class `Variable` is abandon in pytorch version 0.4, the code below may cause something wrong.

```python
inputs, labels = Variable(inputs), Variable(labels.squeeze(1))
```

**Comment by liqi6811** [comment]
thank you zhiqwang, if I do want to input "mask" values, what should be "mask" like? a tensor initialized as zero? if yes, what should be size? Thank you

**Comment by liqi6811** [comment]
Hi zhiqwang, I just tried your suggestion but got error. should I input "mask" values? 

![image](https://user-images.githubusercontent.com/48280760/93478041-f744ef00-f93e-11ea-8bc7-875569fbccf6.png)


![image](https://user-images.githubusercontent.com/48280760/93478207-117ecd00-f93f-11ea-928a-1a7adf246ac5.png)

**Comment by zhiqwang** [comment]
It's just for testing in this formal. If you must include the `mask` in, it's more elegant to use `NestedTensor`. (Just my opinion) And the function `nested_tensor_from_tensor_list` may help you (Also in my own understanding).

https://github.com/facebookresearch/detr/blob/5e66b4cd15b2b182da347103dd16578d28b49d69/util/misc.py#L283-L305

**Comment by liqi6811** [comment]
thank you zhiqwang!

**Comment by fmassa** [comment]
Thanks @zhiqwang for helping answering this question!
